src/gluonts/model/seq2seq/_forking_network.py

```diff
@@ -232,6 +242,62 @@ def hybrid_forward(
         return weighted_loss
 
 
+class ForkingSeq2SeqDistributionTrainingNetwork(ForkingSeq2SeqNetworkBase):
+    # noinspection PyMethodOverriding
+    def hybrid_forward(
+        self,
+        F,
+        past_target: Tensor,
+        future_target: Tensor,
+        past_feat_dynamic: Tensor,
+        future_feat_dynamic: Tensor,
+        feat_static_cat: Tensor,
+        past_observed_values: Tensor,
+        future_observed_values: Tensor,
+    ) -> Tensor:
+        """
+        Parameters
+        ----------
+        F: mx.symbol or mx.ndarray
+            Gluon function space
+        past_target: Tensor
+            shape (batch_size, encoder_length, 1)
+        future_target: Tensor
+            shape (batch_size, encoder_length, decoder_length)
+        past_feat_dynamic
+            shape (batch_size, encoder_length, num_past_feature_dynamic)
+        future_feat_dynamic
+            shape (batch_size, encoder_length, decoder_length, num_feature_dynamic)
+        feat_static_cat
+            shape (batch_size, encoder_length, num_feature_static_cat)
+        past_observed_values: Tensor
+            shape (batch_size, encoder_length, 1)
+        future_observed_values: Tensor
+            shape (batch_size, encoder_length, decoder_length)
+
+        Returns
+        -------
+        loss with shape (batch_size, prediction_length)
+        """
+        assert self.distr_output is not None
+
+        dec_output, scale = self.get_decoder_network_output(
+            F,
+            past_target,
+            past_feat_dynamic,
+            future_feat_dynamic,
+            feat_static_cat,
+            past_observed_values,
+        )
+        distr_args = self.distr_args_proj(dec_output)
+        distr = self.distr_output.distribution(distr_args, scale=scale)
+        loss = distr.loss(future_target)
+        weighted_loss = weighted_average(
+            F=F, x=loss, weights=future_observed_values, axis=1
+        )
+        return weighted_loss
```

Comment:
Maybe this could be handled in an `if distribution ... else ...` since both return `weighted_loss `.

Reply:
addressed